### PR TITLE
Add INAV compatibility via passthrogh

### PR DESCRIPTION
**Pull request by mmosca**
This PR adds support for INAV and allow esc-configurator to work via passthrough.
Main difference is how to detect motor count in INAV vs Betaflight.

### CODE REVIEW

**Comment by mmosca** [comment]
Hopefully, it will close #1

**Comment by mmosca** [comment]
Just a quick update. I was having the same issues with BF. I suspect bad wiring/connector somewhere. I have tried different cable, with no success. Will try direct soldering eventually. But this got me far enough to be able to change motor directions and change settings.

**Comment by mmosca** [comment]
Got new hardware. Seems to work ok. Main issue is that INAV miss reports MOTOR_COUNT, so it will report twice the amount you actually have.

I could not test flashing via INAV, but reading and changing config worked without issue.

**Comment by mmosca** [comment]
Motor count detection has been fixed. :)

Looks good to go.

File: components/SerialDevice.vue [hunk 1]
```diff
@@ -523,11 +523,19 @@ const connectToDevice = async () => {
                             commandsQueue.processMspResponse(result!.commandName, result!.data);
                         }
                     });
-                    await Msp.getInstance().sendWithPromise(MSP_COMMANDS.MSP_MOTOR_CONFIG).then((result) => {
-                        if (result) {
-                            commandsQueue.processMspResponse(result!.commandName, result!.data);
-                        }
-                    });
+                    if (serialStore.mspData.type == 'inav') {
+                      await Msp.getInstance().sendWithPromise(MSP_COMMANDS.MSP_MOTOR).then((result) => {
```
Comment (freasy) [inline components/SerialDevice.vue:527]: i would like a mapping here like `getTypeMotorCommand(type: string): MSP_COMMANDS` that way we dont have duplicate code.

**Review by freasy — CHANGES_REQUESTED** [review]

File: components/SerialDevice.vue (hunk 1 above)
Comment (mmosca) [reply components/SerialDevice.vue:527]: Will do.

**Comment by mmosca** [comment]
Configurator with correct motor count and connected to skystars am32 esc through inav.

image

**Review by freasy — APPROVED** [review]
Code approved, need 2 function approvals

**Comment by mmosca** [comment]
Re-confirmed working with deployed version.

Flash tested.

Downgraded esc1 to 2.10 and back to 2.12
```
11:16:54
initializing...
11:17:01
Connected to device
11:17:01
Sending MSP_API_VERSION...
11:17:01
Got msp data, API_VERSION: 2.5.0
11:17:01
Sending MSP_FC_VARIANT...
11:17:01
Got msp data, TYPE: inav
11:17:01
Sending MSP_BATTERY_STATE...
11:17:01
Got msp data, BATTERY VOLTAGE: 12.2
11:17:01
Sending MSP_MOTOR...
11:17:01
Got msp data, MOTOR COUNT: 4
11:17:06
Sending MSP_SET_PASSTHROUGH...
11:17:07
Sending cmd_DeviceInitFlash...
11:17:07
Sending cmd_DeviceRead...
11:17:08
Sending cmd_DeviceRead...
11:17:08
Sending cmd_DeviceInitFlash...
11:17:08
Sending cmd_DeviceRead...
11:17:08
Sending cmd_DeviceRead...
11:17:09
Sending cmd_DeviceInitFlash...
11:17:09
Sending cmd_DeviceRead...
11:17:09
Sending cmd_DeviceRead...
11:17:09
Sending cmd_DeviceInitFlash...
11:17:10
Sending cmd_DeviceRead...
11:17:10
Sending cmd_DeviceRead...
11:18:30
Sending cmd_DeviceInitFlash...
11:18:30
Sending cmd_DeviceRead...
11:18:30
Sending cmd_DeviceWrite...
11:18:31
Sending cmd_DeviceWrite...
11:18:31
Sending cmd_DeviceWrite...
11:18:31
Sending cmd_DeviceWrite...
11:18:31
Sending cmd_DeviceWrite...
11:18:31
Sending cmd_DeviceWrite...
11:18:32
Sending cmd_DeviceWrite...
11:18:32
Sending cmd_DeviceWrite...
11:18:32
Sending cmd_DeviceWrite...
11:18:32
Sending cmd_DeviceWrite...
11:18:32
Sending cmd_DeviceWrite...
11:18:32
Sending cmd_DeviceWrite...
11:18:32
Sending cmd_DeviceWrite...
11:18:33
Sending cmd_DeviceWrite...
11:18:33
Sending cmd_DeviceWrite...
11:18:33
Sending cmd_DeviceWrite...
11:18:33
Sending cmd_DeviceWrite...
11:18:33
Sending cmd_DeviceWrite...
11:18:33
Sending cmd_DeviceWrite...
11:18:34
Sending cmd_DeviceWrite...
11:18:34
Sending cmd_DeviceWrite...
11:18:34
Sending cmd_DeviceWrite...
11:18:34
Sending cmd_DeviceWrite...
11:18:34
Sending cmd_DeviceWrite...
11:18:34
Sending cmd_DeviceWrite...
11:18:34
Sending cmd_DeviceWrite...
11:18:35
Sending cmd_DeviceWrite...
11:18:35
Sending cmd_DeviceWrite...
11:18:35
Sending cmd_DeviceWrite...
11:18:35
Sending cmd_DeviceWrite...
11:18:35
Sending cmd_DeviceWrite...
11:18:35
Sending cmd_DeviceWrite...
11:18:36
Sending cmd_DeviceWrite...
11:18:36
Sending cmd_DeviceWrite...
11:18:36
Sending cmd_DeviceWrite...
11:18:36
Sending cmd_DeviceWrite...
11:18:36
Sending cmd_DeviceWrite...
11:18:36
Sending cmd_DeviceWrite...
11:18:36
Sending cmd_DeviceWrite...
11:18:37
Sending cmd_DeviceWrite...
11:18:37
Sending cmd_DeviceWrite...
11:18:37
Sending cmd_DeviceWrite...
11:18:37
Sending cmd_DeviceWrite...
11:18:37
Sending cmd_DeviceWrite...
11:18:37
Sending cmd_DeviceWrite...
11:18:38
Sending cmd_DeviceWrite...
11:18:38
Sending cmd_DeviceWrite...
11:18:38
Sending cmd_DeviceWrite...
11:18:38
Sending cmd_DeviceWrite...
11:18:38
Sending cmd_DeviceWrite...
11:18:38
Sending cmd_DeviceWrite...
11:18:38
Sending cmd_DeviceWrite...
11:18:39
Sending cmd_DeviceWrite...
11:18:39
Sending cmd_DeviceWrite...
11:18:39
Sending cmd_DeviceWrite...
11:18:39
Sending cmd_DeviceWrite...
11:18:39
Sending cmd_DeviceWrite...
11:18:39
Sending cmd_DeviceWrite...
11:18:40
Sending cmd_DeviceWrite...
11:18:40
Sending cmd_DeviceWrite...
11:18:40
Sending cmd_DeviceWrite...
11:18:40
Sending cmd_DeviceWrite...
11:18:40
Sending cmd_DeviceWrite...
11:18:40
Sending cmd_DeviceWrite...
11:18:41
Sending cmd_DeviceWrite...
11:18:41
Sending cmd_DeviceWrite...
11:18:41
Sending cmd_DeviceWrite...
11:18:41
Sending cmd_DeviceWrite...
11:18:41
Sending cmd_DeviceWrite...
11:18:41
Sending cmd_DeviceWrite...
11:18:41
Sending cmd_DeviceWrite...
11:18:42
Sending cmd_DeviceWrite...
11:18:42
Sending cmd_DeviceWrite...
11:18:42
Sending cmd_DeviceWrite...
11:18:42
Sending cmd_DeviceWrite...
11:18:42
Sending cmd_DeviceWrite...
11:18:42
Sending cmd_DeviceWrite...
11:18:43
Sending cmd_DeviceWrite...
11:18:43
Sending cmd_DeviceWrite...
11:18:43
Sending cmd_DeviceWrite...
11:18:43
Sending cmd_DeviceWrite...
11:18:43
Sending cmd_DeviceWrite...
11:18:43
Sending cmd_DeviceWrite...
11:18:44
Sending cmd_DeviceWrite...
11:18:45
Sending cmd_DeviceWrite...
11:18:45
Sending cmd_DeviceWrite...
11:18:45
Sending cmd_DeviceWrite...
11:18:46
Sending cmd_DeviceWrite...
11:18:46
Sending cmd_DeviceWrite...
11:18:46
Sending cmd_DeviceWrite...
11:18:46
Sending cmd_DeviceWrite...
11:18:46
Sending cmd_DeviceWrite...
11:18:46
Sending cmd_DeviceWrite...
11:18:47
Sending cmd_DeviceWrite...
11:18:47
Sending cmd_DeviceWrite...
11:18:47
Sending cmd_DeviceWrite...
11:18:47
Sending cmd_DeviceWrite...
11:18:47
Sending cmd_DeviceWrite...
11:18:47
Sending cmd_DeviceWrite...
11:18:47
Sending cmd_DeviceWrite...
11:18:48
Sending cmd_DeviceWrite...
11:18:48
Sending cmd_DeviceWrite...
11:18:48
Sending cmd_DeviceWrite...
11:18:48
Sending cmd_DeviceWrite...
11:18:48
Sending cmd_DeviceWrite...
11:18:48
Sending cmd_DeviceWrite...
11:18:49
Sending cmd_DeviceWrite...
11:18:49
Sending cmd_DeviceWrite...
11:18:49
Sending cmd_DeviceWrite...
11:18:49
Sending cmd_DeviceWrite...
11:18:49
Sending cmd_DeviceWrite...
11:18:49
Sending cmd_DeviceWrite...
11:18:50
Sending cmd_DeviceReset...
11:18:55
Sending cmd_DeviceInitFlash...
11:18:55
Sending cmd_DeviceRead...
11:18:56
Sending cmd_DeviceRead...
11:19:46
Sending cmd_DeviceInitFlash...
11:19:46
Sending cmd_DeviceRead...
11:19:46
Sending cmd_DeviceWrite...
11:19:46
Sending cmd_DeviceWrite...
11:19:47
Sending cmd_DeviceWrite...
11:19:47
Sending cmd_DeviceWrite...
11:19:47
Sending cmd_DeviceWrite...
11:19:47
Sending cmd_DeviceWrite...
11:19:47
Sending cmd_DeviceWrite...
11:19:47
Sending cmd_DeviceWrite...
11:19:47
Sending cmd_DeviceWrite...
11:19:48
Sending cmd_DeviceWrite...
11:19:48
Sending cmd_DeviceWrite...
11:19:48
Sending cmd_DeviceWrite...
11:19:48
Sending cmd_DeviceWrite...
11:19:48
Sending cmd_DeviceWrite...
11:19:48
Sending cmd_DeviceWrite...
11:19:49
Sending cmd_DeviceWrite...
11:19:49
Sending cmd_DeviceWrite...
11:19:49
Sending cmd_DeviceWrite...
11:19:49
Sending cmd_DeviceWrite...
11:19:49
Sending cmd_DeviceWrite...
11:19:49
Sending cmd_DeviceWrite...
11:19:49
Sending cmd_DeviceWrite...
11:19:50
Sending cmd_DeviceWrite...
11:19:50
Sending cmd_DeviceWrite...
11:19:50
Sending cmd_DeviceWrite...
11:19:50
Sending cmd_DeviceWrite...
11:19:50
Sending cmd_DeviceWrite...
11:19:50
Sending cmd_DeviceWrite...
11:19:51
Sending cmd_DeviceWrite...
11:19:51
Sending cmd_DeviceWrite...
11:19:51
Sending cmd_DeviceWrite...
11:19:51
Sending cmd_DeviceWrite...
11:19:51
Sending cmd_DeviceWrite...
11:19:51
Sending cmd_DeviceWrite...
11:19:51
Sending cmd_DeviceWrite...
11:19:52
Sending cmd_DeviceWrite...
11:19:52
Sending cmd_DeviceWrite...
11:19:52
Sending cmd_DeviceWrite...
11:19:52
Sending cmd_DeviceWrite...
11:19:52
Sending cmd_DeviceWrite...
11:19:52
Sending cmd_DeviceWrite...
11:19:53
Sending cmd_DeviceWrite...
11:19:54
Sending cmd_DeviceWrite...
11:19:55
Sending cmd_DeviceWrite...
11:19:56
Sending cmd_DeviceWrite...
11:19:57
Sending cmd_DeviceWrite...
11:19:58
Sending cmd_DeviceWrite...
11:19:59
Sending cmd_DeviceWrite...
11:20:00
Sending cmd_DeviceWrite...
11:20:01
Sending cmd_DeviceWrite...
11:20:02
Sending cmd_DeviceWrite...
11:20:03
Sending cmd_DeviceWrite...
11:20:04
Sending cmd_DeviceWrite...
11:20:05
Sending cmd_DeviceWrite...
11:20:06
Sending cmd_DeviceWrite...
11:20:07
Sending cmd_DeviceWrite...
11:20:08
Sending cmd_DeviceWrite...
11:20:09
Sending cmd_DeviceWrite...
11:20:10
Sending cmd_DeviceWrite...
11:20:11
Sending cmd_DeviceWrite...
11:20:12
Sending cmd_DeviceWrite...
11:20:13
Sending cmd_DeviceWrite...
11:20:14
Sending cmd_DeviceWrite...
11:20:15
Sending cmd_DeviceWrite...
11:20:15
Sending cmd_DeviceWrite...
11:20:15
Sending cmd_DeviceWrite...
11:20:15
Sending cmd_DeviceWrite...
11:20:16
Sending cmd_DeviceWrite...
11:20:16
Sending cmd_DeviceWrite...
11:20:16
Sending cmd_DeviceWrite...
11:20:16
Sending cmd_DeviceWrite...
11:20:16
Sending cmd_DeviceWrite...
11:20:16
Sending cmd_DeviceWrite...
11:20:17
Sending cmd_DeviceWrite...
11:20:17
Sending cmd_DeviceWrite...
11:20:17
Sending cmd_DeviceWrite...
11:20:17
Sending cmd_DeviceWrite...
11:20:17
Sending cmd_DeviceWrite...
11:20:17
Sending cmd_DeviceWrite...
11:20:17
Sending cmd_DeviceWrite...
11:20:18
Sending cmd_DeviceWrite...
11:20:18
Sending cmd_DeviceWrite...
11:20:18
Sending cmd_DeviceWrite...
11:20:18
Sending cmd_DeviceWrite...
11:20:19
Sending cmd_DeviceWrite...
11:20:20
Sending cmd_DeviceWrite...
11:20:21
Sending cmd_DeviceWrite...
11:20:22
Sending cmd_DeviceWrite...
11:20:23
Sending cmd_DeviceWrite...
11:20:24
Sending cmd_DeviceWrite...
11:20:25
Sending cmd_DeviceWrite...
11:20:26
Sending cmd_DeviceWrite...
11:20:27
Sending cmd_DeviceWrite...
11:20:28
Sending cmd_DeviceWrite...
11:20:29
Sending cmd_DeviceWrite...
11:20:30
Sending cmd_DeviceWrite...
11:20:30
Sending cmd_DeviceWrite...
11:20:31
Sending cmd_DeviceWrite...
11:20:31
Sending cmd_DeviceWrite...
11:20:31
Sending cmd_DeviceWrite...
11:20:31
Sending cmd_DeviceWrite...
11:20:31
Sending cmd_DeviceWrite...
11:20:31
Sending cmd_DeviceWrite...
11:20:32
Sending cmd_DeviceWrite...
11:20:32
Sending cmd_DeviceWrite...
11:20:32
Sending cmd_DeviceWrite...
11:20:32
Sending cmd_DeviceWrite...
11:20:32
Sending cmd_DeviceWrite...
11:20:32
Sending cmd_DeviceWrite...
11:20:32
Sending cmd_DeviceWrite...
11:20:33
Sending cmd_DeviceWrite...
11:20:33
Sending cmd_DeviceWrite...
11:20:33
Sending cmd_DeviceWrite...
11:20:34
Sending cmd_DeviceReset...
11:20:40
Sending cmd_DeviceInitFlash...
11:20:41
Sending cmd_DeviceRead...
11:20:41
Sending cmd_DeviceRead...
```

image


image

**Pull request merged**